Itegrate linked Post provider

mathis-m opened the pull request:
Just possibly to use not used so far

mathis-m committed:
split types
add some new linked types for later

mathis-m committed:
Integrate ConfigBuilder

mathis-m committed:
integrate Linked PostInfo Provider

s-bauer commented:
This is blocked due to the merge of #7. Please rebase your changes on top of the `typeRefactor` branch.

mathis-m committed:
Create `LinkedSliverChildBuilderDelegate` Class.
It extends SliverChildBuilderDelegate, so it can be constructed by providing a builder Function of type LinkedWidgetBuilder.
Also a initial LinkedIterator is needed for construction.
The class maps indexes to LinkedIterator and LinkedIterator to Widget;
If a new Index is required the LinkedWidgetBuilder Function is called with the appropriate LinkedIterator.

mathis-m committed:
Create a minimal Clone of the PageView class.
To be constructed the initial LinkedIterator and a LinkedWidgetBuilder Function must be provided.
Those are required by the new `LinkedSliverChildBuilderDelegate` which is used instead of the liverChildBuilderDelegate Class.

mathis-m committed:
Disable the Index Correction until LinkedSliverFillViewport can be committed.

mathis-m committed:
Add Debug LinkedPostPage

mathis-m committed:
extend LinkedIterator mixin

mathis-m committed:
enable LinkedPostPage for debug
